File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prev = {}

        for i, n in enumerate(nums):
            nfind = target - n
            if nfind in prev:
                return [prev[nfind], i]
            prev[n] = i

File: Data Structures & Algorithms/binary-search/submission-3.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int start = 0;
        int end = nums.size() - 1;

        while (start <= end) {
            int ind = start + ((end - start) / 2);

            if (nums[ind] < target) {
                start = ind + 1;
            }

            else if (nums [ind] > target) {
                end = ind - 1;
            }

            else if (nums[ind] == target) {
                return ind;
            }
        }
        return -1;
    }
};
